### DISCUSSION TITLE
I try to install apk, but I cannot do that

### DISCUSSION THREAD

**Discussion by soroshsabz** [infinitepower18/WSA-Sideloader#63] [General]
ITNOA

I try to install apk, but I cannot do that

![image](https://user-images.githubusercontent.com/17947618/220670273-e729be46-cb75-40bf-b898-324e975229be.png)

I check WSA is running (I open Amazon store simultaneously) and enable developer mode

![image](https://user-images.githubusercontent.com/17947618/220670644-f5eecac0-dcd4-4164-af4a-f55f3889164d.png)

How to more debug my problem?

**Comment by infinitepower18** [comment]
It's a WSA bug. Restart your PC and it should work.

**Reply by soroshsabz** [reply]
thanks for fast feedbacking, I try your workaround

where is this bug reported to Microsoft? did you have any link of the bug reported?

**Reply by infinitepower18** [reply]
I already reported it: https://github.com/microsoft/WSA/issues/136

**Reply by soroshsabz** [reply]
I restarted my system, but my problem does not resolved :(

**Reply by infinitepower18** [reply]
That's strange... normally rebooting fixes it for me.

Can you check the developer settings and see if USB debugging is enabled? You can try disabling and enabling it to see if it fixes anything. 

<img width="962" alt="image" src="https://user-images.githubusercontent.com/44692189/220703459-9e976414-4ec6-4f1e-bcdc-75844d46432f.png">

**Reply by soroshsabz** [reply]
![image](https://user-images.githubusercontent.com/17947618/220721394-916b378e-6cae-4006-81c1-4e2cddd2029a.png)


I re-enable, but my problem does not resolved

**Reply by infinitepower18** [reply]
Are you running the latest version of Windows 11?

Try reinstalling WSA, if it still doesn't work then idk what else I can suggest atm. All the APK installers install Android apps on WSA via ADB.

You can also try running the `adb connect 127.0.0.1:58526` command either using the adb downloaded from Google or the one bundled in WSA Sideloader (if you downloaded from MS Store, it's located in `%LOCALAPPDATA%\Packages\46954GamenologyMedia.WSASideloader-APKInstaller_cjpp7y4c11e3w\LocalState\platform-tools`, otherwise you can find it in your installation directory)

**Reply by infinitepower18** [reply]
Since this issue is WSA related there isn't much I can do from my end. If you're still facing trouble after reinstalling and rebooting then open an issue on the [official WSA GitHub](https://github.com/microsoft/wsa) as your situation may be bit different.

**Reply by soroshsabz** [reply]
I use the latest Windows 11 version

```
Edition	Windows 11 Pro
Version	22H2
Installed on	‎11/‎10/‎2022
OS build	22621.1265
Experience	Windows Feature Experience Pack 1000.22638.1000.0

```

```
rm-tools> .\adb.exe connect 127.0.0.1:58526
* daemon not running; starting now at tcp:5037
* daemon started successfully
cannot connect to 127.0.0.1:58526: No connection could be made because the target machine actively refused it. (10061)
PS C:\Users\sooro\AppData\Local\Packages\46954GamenologyMedia.WSASideloader-APKInstaller_cjpp7y4c11e3w\LocalState\platfo
```

**Reply by infinitepower18** [reply]
I think it may be the firewall that is blocking the connection. Open Windows Firewall with Advanced Security and check the inbound, outbound and firewall sections to see if WSA is blocked or not.

<img width="786" alt="image" src="https://user-images.githubusercontent.com/44692189/222913366-870fafe4-2dad-4dda-9a37-2651d79c7d71.png">

I think you should have got a firewall warning when launching WSA for the first time.

**Reply by soroshsabz** [reply]
I have these rules on Outbound Rules

![image](https://user-images.githubusercontent.com/17947618/223786113-32342147-561e-46d8-a478-e63534515d79.png)


I do not think WSA need to inbound rules? Do you agree with me?

**Reply by infinitepower18** [reply]
> I have these rules on Outbound Rules
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/17947618/223786113-32342147-561e-46d8-a478-e63534515d79.png)
> 
> 
> 
> 
> 
> I do not think WSA need to inbound rules? Do you agree with me?

Try adding WSA in inbound rules and see if the problem is solved

**Reply by MustardChef** [reply]
@soroshsabz Try this: 

1. Disable hyper-v
    dism.exe /Online /Disable-Feature:Microsoft-Hyper-V

2. Restart Your PC

3. Reserve port 58526 so hyper-v doesn't reserve it back
    netsh int ipv4 add excludedportrange protocol=tcp startport=58526 numberofports=1

4. Re-Enable hyper-V
    dism.exe /Online /Enable-Feature:Microsoft-Hyper-V /All

5. Restart your PC

**Reply by MustardChef** [reply]
I was having the same issue, and following the steps above fixed the issue stated and other issues releated to ADB and networking in WSA. Its worth a try, I guess.

**Reply by Student414** [reply]
Do not need to disable Hyper-V, just stop NAT service and exclude 58526 port then restart NAT service.
```
PS C:\Users\Student414> net stop winnat

Windows NAT Driver 服务已成功停止。

PS C:\Users\Student414> netsh int ipv4 add excludedportrange protocol=tcp startport=58526 numberofports=1
确定。

PS C:\Users\Student414> net start winnat

Windows NAT Driver 服务已经启动成功。

PS C:\Users\Student414>
```
